rosetta/services/block.go

```diff
@@ -1104,19 +1112,25 @@ func newTransferOperations(
 // on the sender's shard.
 func newCrossShardSenderTransferOperations(
 	startingOperationID *types.OperationIdentifier,
-	tx *hmytypes.Transaction,
+	tx *hmytypes.Transaction, senderAddress ethcommon.Address,
 ) ([]*types.Operation, *types.Error) {
-	// Sender address should have errored prior to call this function
-	senderAddress, _ := tx.SenderAddress()
-	receiverAddress := *tx.To()
 	senderAccountID, rosettaError := newAccountIdentifier(senderAddress)
 	if rosettaError != nil {
 		return nil, rosettaError
 	}
-	receiverAccountID, rosettaError := newAccountIdentifier(receiverAddress)
+	receiverAccountID, rosettaError := newAccountIdentifier(*tx.To())
```

Comment:
any risk of nil pointer?

Reply:
Don't think so since cross-shard transactions need a destination address and some verification is done to the tx prior to calling this internal function. That said, better safe than sorry, so I'll review and add some checks for all uses of `*tx.To()`.